data_structures/avltree.cpp

```diff
@@ -6,38 +6,52 @@
  * \warning This program is a poor implementation and does not utilize any of
  * the C++ STL features.
  */
-#include <algorithm>
-#include <iostream>
-#include <queue>
+#include <algorithm>  // for std::max
+#include <iostream>   // for std::cout
+#include <queue>      // for std::queue
```

Comment:
```suggestion
#include <algorithm>  /// for std::max
#include <iostream>   /// for IO operations
#include <queue>      /// for std::queue
```

Reply:
I agree with missing `/`. I do not agree with `IO operations`. The line `#include <algorithm>  /// for std::max` means that the `algorithms` are included because of `std::max`. One way of checking if this header is really necessary is to look for all occurrences of string `std::max`. If there are at least two matches, then it is needed. The line `#include <iostream>   /// for IO operations` does not have this useful property.